effekt/jvm/src/test/scala/effekt/PolymorphismBoxingTests.scala

```diff
@@ -0,0 +1,286 @@
+      val oldBound = bound
+      bound = ids ++ bound
+      val res = f
+      bound = oldBound
+      res
+    }
+    def withBinding[R](id: symbols.Symbol)( f: => R ): R = withBindings(List(id))(f)
+
+    override def pure: PartialFunction[Pure, Pure] = {
+      case Pure.ValueVar(id, tpe) => Pure.ValueVar(rewrite(id), tpe)
+    }
+    override def defn: PartialFunction[Definition, Definition] = {
+      case Definition.Def(id, block) =>
+        Definition.Def(rewrite(id), rewrite(block))
+      case Definition.Let(id, block) =>
+        Definition.Let(rewrite(id), rewrite(block))
+    }
+
+    override def stmt: PartialFunction[Stmt, Stmt] = {
+      case core.Scope(definitions, rest) => withBindings(definitions.map{
+          case core.Definition.Def(id, _) => id
+          case core.Definition.Let(id, _) => id
+        }){
+        core.Scope(definitions map rewrite, rewrite(rest))
+      }
+      case core.Val(id, binding, body) => withBinding(id){
+        core.Val(rewrite(id), rewrite(binding), rewrite(body))
+      }
+      case core.State(id, init, reg, body) => withBinding(id){
+        core.State(rewrite(id), rewrite(init), rewrite(reg), rewrite(body))
+      }
+    }
+    override def block: PartialFunction[Block, Block] = {
+      case Block.BlockVar(id, tpe, capt) => Block.BlockVar(rewrite(id), tpe, capt map rewrite)
+      case Block.BlockLit(tparams, cparams, vparams, bparams, body) =>
+        withBindings(cparams ++ vparams.map(_.id) ++ bparams.map(_.id)){
+          Block.BlockLit(tparams, cparams map rewrite, vparams map rewrite, bparams map rewrite,
+            rewrite(body))
+        }
+    }
+    def rewrite(id: symbols.Symbol): symbols.Symbol = {
+      if(bound.contains(id)){
+        names.idFor(prefix ++ (bound.length - bound.indexOf(id)).toString)
+      } else id
+    }
+    def rewrite(b: core.Param.BlockParam): core.Param.BlockParam = b match {
+      case core.Param.BlockParam(id, tpe) => core.Param.BlockParam(rewrite(id), tpe)
+    }
+    def rewrite(b: core.Param.ValueParam): core.Param.ValueParam = b match {
+      case core.Param.ValueParam(id, tpe) => core.Param.ValueParam(rewrite(id), tpe)
+    }
+
+    def apply(m: core.ModuleDecl): core.ModuleDecl = m match {
+      case core.ModuleDecl(path, imports, declarations, externs, definitions, exports) =>
+        core.ModuleDecl(path, imports, declarations, externs, definitions map rewrite, exports)
+    }
+  }
+
+  def assertTransformsTo(input: String, expected: String): Unit = {
+    val pInput = core.CoreParsers.module(input, names) match {
+      case Success(result, next) => result
+      case nosuccess: NoSuccess => fail(nosuccess.toMessage)
+    }
+    val pExpected = core.CoreParsers.module(expected, names) match {
+      case Success(result, next) => result
+      case nosuccess: NoSuccess => fail(nosuccess.toMessage)
+    }
+    implicit val pctx: core.PolymorphismBoxing.PContext = new PolymorphismBoxing.PContext(List())(using context)
```

Comment:
In Scala 3 this would look a bit different. I can't remember myself and always have to look up the syntax, though :(

Reply:
It's `given T = <some expression of type T>`. Fixed.